### ISSUE TITLE
New Website for KVIrc

### ISSUE THREAD

**Issue by RCMainak** [kvirc/kvirc-www#20]
Hello everyone! As discussed on thr IRC I opened up this issue here to track things that needs improvements/attention. If you haven't saw the new prototype of our website here is the [link](http://rcmainak.github.io/kvirc). Please provide your valuable feedback. I'll put a list of things needed to be fixed : 
- [ ] Title
- [x] ~~First public KVIrc releases were around **1998** (maybe even earlier..). Correct this info~~
- [x] ~~The (historic) team is very large. Added to github,~~ Still need to make a separate page???
- [x] ~~Add a section for themes & scripts maybe.~~
- [x] ~~We should direct people to the irc channel, to github or to the mailing list. No need for a send-mail form.~~
- [x] ~~Add a contact button in the menu section~~
- [ ] Make easy reach places for the real information like building info, pointers to releases, FAQs etc.
- [ ] Make something for JS disabled browsers.
- [ ] Add image of KVIrc on the Explore section
- [ ] The Download links or buttons should do something. We can either start the download of specific installer according to the users platform. Yes, I can detect user's OS. or we can redirect to a download page which again may contain a specific installer depending on the OS or may contain all the installer's.
- [ ] Fix typos and anything which will make the website better.
- [ ] Add a background image in the home section
- [ ] Make the View Demo button work. On clicking on the button user's will be redirected to a youtube video which is nothing but a tutorial on how to setup the IRC client.
- [ ] Make it a little bit less "marketing-oriented". We don't need to sell a product
- [ ] Add XP to the list of supported platform for KVIrc 5.0

Now it's your turn. Blow up this issue with your creative ideas. All the great will be included in the list above to make things easier to track. Thank you

**Comment by extioStorm** [comment]
"The Chat cool people use" is the slogan of the irc client IceChat. http://www.icechat.net/site/

**Comment by wodim** [comment]
There's literally nothing I like about it. I'm sorry.

**Comment by un1versal** [comment]
> There's literally nothing I like about it. I'm sorry.

With these comments should come suggestions, like links to sites and visuals you like, something other than this put down.

**Comment by pragmaware** [comment]
Interesting starting point. Looks modern, which is something we probably need.
Random ideas:
- I'd try to make it a little bit less "marketing-oriented": It's good to make the user curious and interested but we don't need to sell a product :)
- First public KVIrc releases were around 1999 (maybe even earlier.. I just can't remember)
- Maybe having a "shiny" frontpage might be a good idea. But then we need an easily reachable place for the real information like building info, pointers to releases, FAQs etc.
- The (historic) team is very large, it probably deserves a page on its own.
- We should direct people to the irc channel, to github or to the mailing list. No need for a send-mail form.

**Comment by RCMainak** [comment]
@un1versal In IRC @wodim provided a few example such as and clarified a few issues [HexChat](https://hexchat.github.io/) & [Irssi](https://irssi.org/). He didn't liked it because :
1. It's too much fancy
2. I used a template
3. I used a iframe element which dynamically loads up current date of a specific timezone from the server and uses http.
4. The most controversial one list of contributor.

and these are my thoughts with respect to each of the points :
1. All the example of simpler example have their uniqueness in their website. We should have uniqueness. A fancier website will grab a few more users in my opinion. The client is great why else do you think we don't have a better user base.
2. Yes! Of course I used a template how else you roll out a website prototype within a few hours. In my opinion this what Open Source means, reuse others great work, appreciate the good work and concentrate on the more important work.
3. Well, there is no better way to do what I wanted. All though there is a https version of it available or maybe I can just simply make it static. It's not a big deal, All I wanted to do was make things easier to maintain. 
4. I simply just want to pay tribute to the former contributors and respect to the current contributors and maybe a bit of motivation to the future contributors. 

I'm open to all type of modifications and suggestion i'm just trying to help. Please let me help by discussing what we really need :)

**Comment by pragmaware** [comment]
Things I like:
- Modern and clean look. Not necessairly this one but something that has large buttons, organized panels without too much clutter and a clean font is certainly a good idea.
- Full browser width.
- The idea of a landing page with "quick download" and probably a brief presentation (a screenshot and a list of features) like the "Explore" section.

Then "behind" maybe it's better to have some kind of static layout with a menu and separate pages for content.

**Comment by un1versal** [comment]
This was a good start, too much marketing based (we should project a real, fun + organic image not some cold corporate like robots following some recipe to success)

---

Essentially imo we should place focus on introducing the client, get people to download it/install and can easily access documentation/wiki/addons/scripts/themes (what pragma said on steroids)
- attract users and developers (not crazy people we have enough of that)
- your name on the copyright footer not the first thing people see (this isnt about you its about KVIrc)

__ Footer __ something like

<p align="center">
  <b>KVIrc © 2016 The KVIrc Development Team</b> Website by: RCMainak
</p>

maybe show forks/stars like in http://jekyllthemes.org/ ?? something else like current commits?

---

IRSSI has a simple site with access to everything (but no vavavoom), + its ancient ideas of presenting that information, imo they loose the person interest fast (faster than hexchat website I was outta there in no time always) with many calls to action (no vavavoom). but landing page is effective enough... but still (no vavavoom)
We dont want to copy them, we want to be better, right? +++ **vavavoom** :tongue: 

I dont care if it you use a theme but needs to be right and not all over the place (the top this site you done is too all over the place -- not very usability friendly + dont like the color scheme used, theres no **vavavoom**) some customization maybe necessary to become **unique**.
- You have around 7 seconds or less to capture interest from visitors, dont waste them on wrong things and gimmicks

Personally I prefer dark clean designs with a central flow personally (easier to read?) + like **some ideas** in 
- https://panossakkos.github.io/personal-jekyll-theme/  
  - I love the (typing text) reminds me of typing somewhere like in KVIrc... 
  - top menu (should be bigger logo that gets smaller when moving away) with **big navigation**

--- Fancy **!=** Vavavoom ---

But some ideas that come to mind
- should work on mobile devices as well.
- Top navigation - where we have the program name/logo at top left + navigation (always visible and not on far right) I like this top/menu + navigation style http://y7kim.github.io/agency-jekyll-theme/
- landing page with program screenshots where the live typing text is ( the live typing text could be a teaser for the features) **no screenshot carousel** yuk 
- **no center aligned text** its not a poem. 
- Should have a news page? **+** a simple way to add new news. (no separate blog, no wordpress, no public blog platforms with heavy cms)
- link to wiki, documentation (what pragma said)
- not crazy about the contributors page but if we have one its either everyone or no one and should take least focus away from whats important. **wiki? is maybe a good place for showing team + historic contributors** ??
- some search bar always on?
- **no redirection to anywhere** if we have a video **WE DONT** want people to leave site **unless its to goto KVIrc Wiki/github**) see http://t413.com/SinglePaged/#video
- a main download page (not all OS has a downloadable installer like Linux,Unix )
  The problem with current download page is pages within pages we need simple way to access all downloads??
- If we have addons/scripts/themes -- **we should maybe** Load addons - scripts / Themes from their repos (automatically) **currently its a huge pain** to add anything to website and its all broken)

**Comment by un1versal** [comment]
mailing lists? does anyone read those and reply? My guess looking at current trend is **NO**
Its hard enough to get people involved on IRC, GitHub issues, PR feedback.

**Comment by DarthGandalf** [comment]
This new website currently doesn't work at all with JS disabled.

**Comment by RCMainak** [comment]
_UPDATE_ : rolled out v2 with improved content and navigation. Played a bit with the layout and I think everyone will like the results. Included navigation options like Add-ons, wiki etc.

**NOTE** : for now I'm concentrating on the content & functionalities of the website. I'm not worried about color scheme, animation or  arts. I have those in mind and on my to-do list. I'll come back to JS disabled after the enabled version is finalized. i don't have much experience with it so could use some help :p

Go check out the v2 & let me know if it have all the necessary information or not.

Things needs clarification :
@un1versal It was never about me and always abut KVIrc. The final release will not have my name on the website. :p I included my name to use it as a stamp of under development and I'm the one who is doing the dev work. :D hope no one feels bad about it. Also the footer suggestion is a great one i was confused about what to write there. 

I'm overwhelmed by the feedback so far. I thought no one will take  the time to provide feedback. I put this webpage together within two or three hours while i was on the IRC and taking care of other stuff. i knew it was not perfect but someone should take the liberty to act. Keep the feedback coming I'm trying hard to make it a website that KVIrc deserves.

~~**_KNOWN ISSUE :_** The Wiki link does not work on left click of the mouse but works if you right click on it and open in new tab. :( Not sure where the mess is happening but I'll dig that shit up... anyone can help ;)~~

**Comment by un1versal** [comment]
mmm can you live load the scripts/addons/themes? i.e. immediatly get the contents of whatever is on repos displayed even if something else is added without touching website code?

wiki link up is a + and a good start to get access to all documentation, idk in my head this vision of eventually that would be nice, if the whole wiki/website/themes/addons would be seemless navigating experience without being overwhelming, but not sure if Im dreaming or what or what others think.

Im guessing the vavavoom comes much later...

**Comment by un1versal** [comment]
Also see https://github.com/kvirc/KVIrc/pull/1894 XP will be supported at least for 5.X

**Comment by un1versal** [comment]
Those things your crossing off needs lots of work from where I sit. Ill shut up now lol

**Comment by RCMainak** [comment]
Well, it's matter of prespective. Please help me by being specific. I'm open to suggestions provide feedback of every change. Also if anyone want to help I'm also open to that. Please provide information about what needs improvement. 

> idk in my head this vision of eventually that would be nice, if the whole wiki/website/themes/addons would be seemless navigating experience without being overwhelming, but not sure if Im dreaming or what or what others think.

cross it off your list too much information will be overwhelming and less information will make it hard to navigate, I'm afraid :(

**Comment by RCMainak** [comment]
I need some one to write up features of KVIrc I copied it from the old outdated one. I don't know the ins & outs of KVIrc. So someone help me. 

Where are we going to release the news?? on the website or somewhere else!! I can make a space in the webpage but that will be hard to maintain maybe... 

Do we still need improvements with the navigation bar??

**Comment by HelLViS69** [comment]
Hello,
I tried to see the demo but from mobile I just see a list of colors, no
contents..

Just a little point: "uniqueness" and "template (of someone else) are not
friends

HelLViS69

**Comment by un1versal** [comment]
I was quite specific on my comments I even gave you links to what I meant. Especially navigation wise.

What your crossing off that I said needs work is harder, tbh personally dont want to comment on each tiny change you do or waste a long time on detailed feedback, its annoying to everyone.

Twitter or Facebook Google + or any of that similar type _unsocial_ websites wiil never be for me to write news. **this needs to be simple and easy** the next suggestion is simple but not easy,
else no one will do it... (take now as an example its mostly tumbleweeds) if we can throw a GitHub mardown file on a website directory and that turns into HTML news that would be ideal (Im assuming we dont want CMS? some really light mardown orientated CMS that tunrs into HTML would be ideal) similar for pictures on those news.. chuck a pic on a folder and refer on markdown (not too dissimilar from current)

In fact that can go for themes and addons/scripts a mardown file with text/formatting and any images to go on some other folder.... so entries are populated on site? that is if we cant do that scrapping from the repos in real time or something downloading from repos not sure it will work with current system. I get off right here, Ive no idea about that, I dont understand it and not sure if I want to either.

what @HelLViS69 said is tru but a template can be easily modified by ppl who know what they are doing, making them unique and more tailored to the project needs else doing it from scratch well, this job is not easy. @RCMainak important is that you dont get discouraged, it shouldn't be hard to replace the old website.

@HelLViS69 was the guy who did old website so maybe you guys should discuss it?

I still would like a wiki/website seemless transition though.

**Comment by RCMainak** [comment]
Ops! I forgot about that link to the template. Trust me I had that template sitting on my computer even before I started working on the KVIrc website. :smile: Okay, if everybody agrees upon this [template](http://y7kim.github.io/agency-jekyll-theme/) I can make it work like a charm... :smiley: I thought it's a bit dark for a official IRC client webpage. :tongue: 

Share your views on that theme guyz...

**Comment by DarthGandalf** [comment]
This template looks like some paid service provider, not like software
project. :-(

2016-03-03 15:05 GMT+00:00 Mainak Roy Chowdhury notifications@github.com:

**Comment by un1versal** [comment]
that template I like only the menu size options (not style) at the top and way it behaves when you scroll down.

**Comment by RCMainak** [comment]
@un1versal  Ok, if I make all the animation that happens when you scroll disappear and make the menu size small when you scroll down will that make our current prototype a better choice :question::question::question:  Please, suggest a color scheme... :smiley: 

@DarthGandalf that template is licensed under Apache most probably version 2.0 I don't dig up my files to ensure that but I'm pretty sure it's under Apache. Also we can change texts to make it look like a non-profit product organization :tongue:

**Comment by un1versal** [comment]
I think @darthgandalf comment means the look of the template not the content, but idk.

> @un1versal Ok, if I make all the animation that happens when you scroll disappear and make the menu size small when you scroll down will that make our current prototype a better choice :question::question::question: Please, suggest a color scheme... :smiley: 

Not small, we want big (pragma and me read comments)and we already discussed colors on IRC
amd menu should not disappear imo (read my earlier comments on 2nd reply here)

I prefer dark websites but idk Ill will go with what the majority say on this or if we can have a dark/light themes then even better

**Comment by DarthGandalf** [comment]
@RCMainak I wasn't talking about license of template... @un1versal right.

**Comment by un1versal** [comment]
Possible color palette to help this process...

And again I favor dark background with light text but that palette will fit both (not a fixed in stone palette its just a suggestion)

![kvirc_website_palette](https://cloud.githubusercontent.com/assets/3521959/13526995/be885f50-e203-11e5-9865-c7c3fcee9062.png)

Purple/Yellow inspired by the KVIrc logo

As for template themes... Its great to get a quick idea of how it **_could**_ look, but the overall feel is that the theme choice for the prototype is to restrictive to even remotely suit **the website that KVIrc deserves** as RCMainak so fondly put it.

IMO its important to get the **look**, **style** more or less down, we know what content this website should have and that can be tweaked.
- Website/Wiki in a seamless manner (ideally use a GitHub mardown style CMS (for wiki side) that turns into HTML via this Jekyll business)

**Website** for introducing KVIrc and getting people to download/install it where to get news/goto GitHub
**Wiki** To put all documentation into.

Both should have same look & feel imho

**Comment by staticfox** [comment]
Software dev team and web dev team links are backwards

**Comment by RCMainak** [comment]
@staticfox nice one... lol I'll fix it.

@un1versal after yesterdays discussion I put together I much better version of color combo, I'm not a color person :tongue: let me know how is it http://i1379.photobucket.com/albums/ah158/jockr2/KVIrc/Capture3_zpsmc78hwfo.png 
I'm sure it needs improvement lol

@HelLViS69 I forget ask which kind of mobile are you using!!! I checked  the prototype from 
Device : moto e 2nd gen,  OS :  lolipop,  Browser : Chrome
and it works. My guess is you're using a mobile which doesn't support HTML5 or JS. let me know...

**Comment by un1versal** [comment]
Ive put forward a color palette that if you can conjugate those colors properly it should look fine.

The trick is to use a good contrast, yellow on purple isnt quite right looking imo.

Under the right circumstances I would do website mockups, but Im doing other stuff that if I dont finish now Ill get sidetracked with this website thing which is going to consume a huge chunk of my time.
If I dwelve into the code and chnage it myself I wont come up for air until its finished and then I end up getting lost with deving the website which I dont want.

So with the feedback you going to have to make do. but that template on prototype has to die. its not right for the project. sorry me no like and cant visualize something better looking at it, its uninspiring.

**Comment by HelLViS69** [comment]
> @HelLViS69 I forget ask which kind of mobile are you using!!! I checked the prototype from
> Device : moto e 2nd gen, OS : lolipop, Browser : Chrome
> and it works. My guess is you're using a mobile which doesn't support HTML5 or JS. let me know...

Chrome (last version) on Android 4.2.2

HelLViS69

**Comment by un1versal** [comment]
heres a brain fart for the header type (small size) when you scroll away

![untitled-1](https://cloud.githubusercontent.com/assets/3521959/13528049/a86edafc-e20b-11e5-8523-fff1b4a9e7d2.png)

**Comment by HelLViS69** [comment]
2016-03-04 14:19 GMT+01:00 un1versal notifications@github.com:

> heres a brain fart for the header type (small size) when you scroll away

Personally I don't like yellow next to purple, but it's just my opinion

HelLViS69

**Comment by un1versal** [comment]
So you dont like the KVIrc logo for same reason? :grinning: 

Doesnt have to be in any case (done as suggestion for @RCMainak) to get a jump start), can be all one color like #f2f2f2 or #b2ba1d or one of the palette colors that has enough contrast against the dark bg

One day I'll find the patience to redesign the KVIrc logo anyway.

**Comment by un1versal** [comment]
> Software dev team and web dev team links are backwards

Didn't even know there was a web dev team... but I see this goes to contributors.
IMO we shouldnt do that for the website or the addon/themes

The people who commit code in addons/themes repos arent necessarily the authors of said code

Theres only 1 team... The KVIrc Development Team.

if we doing anything like this we have some small reference to links that are called contributors Im not even sure how to make this relevant enough in a website to introduce and allow people to download the software.

## Main goals
- read news about whatever related. (doesnt have to be a whole section if done right).
- Introduce KVIrc get people excited
- Get/show visitors how to download or compile KVIrc
- Show those who use kvirc access to documentation (wiki)
- Direct people to source code  
  ...  
  ...  
  ...
- **Team** reference the contributors behind the lovely project **In the footer** with links  
  **all non crucial** information should go in footer to keep main website simple.

Both wiki and website should have same style / design and be able to easily add more content to wiki and news part.

Sorry @RCMainak if I was putting you down, you should have enough feedback to build a website that does the job required for the software, Im looking forward to see what you achieve with that feedback.

**Comment by un1versal** [comment]
@RCMainak so you gave up?

**Comment by RCMainak** [comment]
Well, I haven't yet. But, I'm looking for a good reason not to give up. The requirements/expectations are too high compared to my knowledge. We want a responsive design which will look good and feel seamless but need to support JS disabled user. Choosing a good color combination seems really hard for me I'm not very good color person. We want a website but can't make it marketing oriented, I don't even know what marketing means. Everything on the web seems marketing oriented to me expect wikipedia but it can be used for marketing purposes(maybe). No one is working on/don't have the time to work on a website. I used a template to make the initial work easier for me, the effort got misunderstood. I know anyone can download a template & do the work/joke I did but no one is even doing that. The whole decision to work on a website was a bad decision on my end.

I'm not experienced as most of contributors and I respect all regardless of the matter they encouraged me or discouraged me. The feedbacks were great but, there was some misunderstanding on my end. After all the effort it appeared to me maybe I'm wasting time of the community member and mine as well. Maybe I should never thought of starting this issue in the first place it could have saved times of all the people who tried to help me. Also there are some wired bug reports which I can't figure out how to reproduce. It seems too much work for a single person. I'm still confused what should I do next; leave with the respect remaining with me or further humiliate myself.

**Comment by un1versal** [comment]
marketing http://www.merriam-webster.com/dictionary/marketing

Well, ive tried to outline outlined the basic needs in  https://github.com/kvirc/kvirc-www/issues/20#issuecomment-192678729 (idk if everyone agrees or not)
The problem is that templates arent a great something when trying to achieve a website that suits the needs but its good to make a quick website to try illustrate what youre trying to achieve

I sympathize with you, doing a website for anyone is never easy the relation between coder/customer has never been simple or easy.

tbh you cant see this as a humiliation of yourself, no one is judging you and if you feel any comments by anyone caused you to feel this, please say so.
The process of doing something youre not 100% a master requires trial and error, more error than success and failure is only when you give up and none of it is humiliation.

anyway... you call, its not going to be easy but I have some faith with some guidance you can achieve it.

**Comment by RCMainak** [comment]
Well, I know meaning of the word marketing. I meant to say that I don't know how marketing works & all the details :tongue:

**Comment by un1versal** [comment]
You dont need to sell a product, but in some way we need to show some details about the product we just dont need to be all corporate and **stiff** about it, we should be casual and simple.

**Comment by DarthGandalf** [comment]
We indeed have lots of conflicting/ambiguous requirements, because
almost everyone has an own opinion.

**Comment by un1versal** [comment]
hence why we should come up with a list or requirements everyone agrees.

**Comment by RCMainak** [comment]
Can we agree on a list of things we need if we could it'll be a good achievement as a community plus it'll make things much easier to work on.
Also, I respect everyone's opinion and believe that everyone should have equal right to speak in an open source community.

**Comment by un1versal** [comment]
see https://github.com/kvirc/kvirc-www/issues/21

**Comment by un1versal** [comment]
from the last few PM's on IRC from @RCMainak the general impression is that this was a lot of hot wind and that nothing was going to come of it.

@RCMainak expressed huge regrets for starting ANY work and expressed somewhat immature opinions and views that hinted at deleting these threads related to website development and changing his nick and disappearing from our radar.

My view on this is simple. This was never going to be an easy job that could be achieved with little effort or less dedication and sadly not everyone has what it takes to go beyond hot air and templates.

Sad it came to this, but now its time to close this thread as its obvious this wont go anywhere with this person.

TL;DR
We have #21 to generate team discussion and hopefully agree on what we need from a website so if anyone ever wishes to take that job on, has clear goals and instructions.

**Comment by RCMainak** [comment]
Thank you @un1versal for letting the community know about the situation. I'm a lot less experienced from anyone of you maybe that's why many things don't make sense to me. I need real help not only words. It is my fault; I'm not blaming anyone else. By the way I changed my mind about disappearing, all though no one cares about it here lol.

Forgive me for commenting here after this issue is marked as closed(me & my immaturity).
